var/spack/repos/spack_repo/builtin/packages/mgard/package.py

```diff
@@ -10,68 +10,90 @@
 
 class Mgard(CMakePackage, CudaPackage):
     """MGARD error bounded lossy compressor
-    forked from https://github.com/CODARcode/MGARD with patches to support Spack"""
 
-    # This is a research compressor with a fast evolving API.  The fork is updated
-    # when releases are made with minimal changes to support spack
+    For versions up to 2023-12-09, uses a fork from https://github.com/robertu94/MGARD
+    with patches to support Spack. For versions after 2023-12-09, uses the upstream
+    repository at https://github.com/CODARcode/MGARD directly from the master branch.
+    """
+
+    # This is a research compressor with a fast evolving API. Historical versions
+    # use a fork with minimal changes to support spack, while new versions use upstream master.
 
     homepage = "https://github.com/CODARcode/MGARD"
-    git = "https://github.com/robertu94/MGARD"
+    git = "https://github.com/robertu94/MGARD"  # Fork for existing date- versions
+
+    def url_for_version(self, version):
+        """For newer versions, use upstream MGARD repository"""
+        version_str = str(version)
+        if not version_str.startswith("date-"):
```

Comment:
can we just compare the date string with re.match here instead of adding the prefix?

```suggestion
        if re.match(r"[0-9]{4}-[0-9]{2}-[0-9]{2}", version):
```

Reply:
The issue is that if compare dates with semantic versioning, dates always win since they have a bigger value ex: `2024-` vs `1.` .

Reply:
Hm, yeah I see.

Instead of date I think it would be good to note that these are tags on the fork for spack compatibility vs the upstream then.

`compat-<date>`?

Also, if you want the 1. version to be preferred you need to drop the preferred label from the latest date version.

Reply:
It looks like the revisions have been changed to use `compat-`, but this is still checking `date-`.

```suggestion
        if not version_str.startswith("compat-"):
```